### PR TITLE
COCOS-100 - Isolate the algorithm execution from Agent

### PR DESCRIPTION
**Pull request by danko-miladinovic**
# What type of PR is this?

This PR fixes the following issues: 
#100 
#65 
## What does this do?
This PR creates a Linux Namespace for the algorithm executable so that it cannot interfere with the Agent or the kernel.

## Which issue(s) does this PR fix/relate to?
Resolves:
#100 
#65 

## Have you included tests for your changes?

The tests will be included.
## Did you document any new/modified feature?
The documentation will be updated in a separate PR to the cocos-docs repository.

### CODE REVIEW

File: agent/service.go [hunk 1]
```diff
@@ -272,3 +296,35 @@ func run(algoContent, dataContent []byte) ([]byte, error) {
...
+	if err := syscall.Chdir("/"); err != nil {
+		fmt.Println("failed to chdir to /: ", err)
+		os.Exit(1)
+	}
+
+	// if err := syscall.Mount("proc", "/proc", "proc", 0, ""); err != nil {
+	// 	fmt.Println("failed to mount /proc: ", err)
+	// 	os.Exit(1)
+	// }
+
```
Comment (dborovcanin) [inline agent/service.go:325]: Remove unused code.

File: agent/service.go [hunk 2]
```diff
@@ -51,6 +52,10 @@ var (
 	errStateNotReady = errors.New("agent not expecting this operation in the current state")
 	// errHashMismatch provided algorithm/dataset does not match hash in manifest.
 	errHashMismatch = errors.New("malformed data, hash does not match manifest")
+
+	// new root directory
```
Comment (dborovcanin) [inline agent/service.go:56]: No need for this comment.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: agent/service.go (hunk 1 above)
Comment (danko-miladinovic) [reply agent/service.go]: Done

File: agent/service.go (hunk 2 above)
Comment (danko-miladinovic) [reply agent/service.go]: Done

**Review by dborovcanin — COMMENTED** [review]
Is this one ready?

**Comment by danko-miladinovic** [comment]
> Is this one ready?

There is a problem with `initramfs`. It cannot perform the operation `pivot_root`. I  am checking for a solution.

**Comment by drasko** [comment]
Closed in favor of Docker and Wasm runtimes

**Pull request closed without merging**